pkg/serial/handshake.go

```diff
@@ -162,9 +162,26 @@ func HandshakeClient(ctx context.Context, conn net.Conn, debug bool) error {
 	return nil
 }
 
-func readMultiple(conn net.Conn, b []byte) (int, error) {
+func readMultiple(ctx context.Context, conn net.Conn, b []byte) (int, error) {
 	if runtime.GOOS != "windows" {
-		return conn.Read(b)
+		type ioret struct {
+			n   int
+			err error
+		}
+		c := make(chan ioret, 1)
+
+		go func() {
+			n, err := conn.Read(b)
+			c <- ioret{n, err}
+			close(c)
+		}()
+
+		select {
+		case ret := <-c:
+			return ret.n, ret.err
+		case <-ctx.Done():
+			return 0, ctx.Err()
```

Comment:
unless I miss something it's more involved though as only way to cancel the read in rawconn is sending an error over error channel. So we can create a context with deadline in SetReadLine and write timeout err to that channel but that also means that a successful read needs to cancel the context.

To be clear, do you prefer SetReadDeadline to be implemented for this PR or are you OK postponing it to some other day?